### PR TITLE
Enable some code analyze rules

### PR DESCRIPTION
**Pull request by WeihanLi**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Refactoring
- [x] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

1. use code analyzer rule to enforce some best practices
2. `IDE0005` for code style, remove unnecessary namespace using to keep code clean
3. `IDE0040` for code style, always add accessibility modifiers to declare accessibility explicitly
4. `CA1852` for performance, always sealed a type if applicatable
5. Apply license header

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Enable `CA1852`/`IDE0040`/`IDE0005` code analyzer rules         |
| 🇨🇳 Chinese |  启用 `CA1852`/`IDE0040`/`IDE0005` 代码分析规则          |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/core/Helpers/RenderFragmentHelper.cs [hunk 1]
```diff
@@ -1,7 +1,23 @@
...
+After:
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Components;
+*/
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
```
Comment (ElderJames) [inline components/core/Helpers/RenderFragmentHelper.cs:17]: Is that need to be clean?

File: components/core/PresetColor.cs [hunk 2]
```diff
@@ -12,6 +18,17 @@ public enum PresetColor {
...
+        Before:
+                Lime    
+            }
+        } 
+        After:
+                Lime
+            }
+        }
+        */
+        Lime
```
Comment (ElderJames) [inline components/core/PresetColor.cs:32]: And this?

File: components/input/Search.razor.cs [hunk 3]
```diff
@@ -1,8 +1,25 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Components;
+*/
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
```
Comment (ElderJames) [inline components/input/Search.razor.cs:19]: This

File: components/slider/SliderMark.cs [hunk 4]
```diff
@@ -1,5 +1,22 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Components;
+*/
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Components;
```
Comment (ElderJames) [inline components/slider/SliderMark.cs:19]: This

File: components/tag/Tag.razor.cs [hunk 5]
```diff
@@ -1,9 +1,27 @@
...
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.RegularExpressions;
+*/
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
```
Comment (ElderJames) [inline components/tag/Tag.razor.cs:18]: This

File: components/timeline/TimelineItem.cs [hunk 6]
```diff
@@ -1,5 +1,23 @@
...
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Components;
+*/
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
```
Comment (ElderJames) [inline components/timeline/TimelineItem.cs:18]: This

File: components/slider/SliderMark.cs (hunk 4 above)
Comment (WeihanLi) [reply components/slider/SliderMark.cs:5]: Seemed an issue for `dotnet-format`, could be cleaned

Comment (ElderJames) [reply components/slider/SliderMark.cs:5]: Thanks, we also want to fix all warnings :)

File: site/AntDesign.Docs.Build.CLI/KnownChineseTranslations.json [hunk 7]
```diff
@@ -4193,9 +4193,6 @@
   "Toggle the border style.": {
     "Translated": "切换边框样式。"
   },
-  "Whether to enable virtualization feature or not, only works for .NET 5 and higher": {
```
Comment (WeihanLi) [inline site/AntDesign.Docs.Build.CLI/KnownChineseTranslations.json:4196]: Seems like a merge issue, should be revert

**Review by ElderJames — APPROVED** [review]
Thank you!

**Pull request merged**